PANDORA/Pandora/Modelling_functions.py

```diff
@@ -122,94 +122,101 @@ def predict_anchors_netMHCIIpan(peptide, allele_type, verbose=True):
-            all_netMHCpan_alleles.append(line.replace('\n', ''))
-
-    # Format the alles to netMHCIIpan readable format
-    target_alleles = [i.split('-')[-1].replace('*', '_') for i in allele_type]
-
-    # The DQ and DP alleles only function in pairs in netMHCIIpan, which we cannot match from our alleles
-    # So take the first 3 partially matched allele combinations
-    for i in target_alleles:
-        if 'DRB' not in i:
-            target_alleles = target_alleles + [al for al in all_netMHCpan_alleles if i.replace('_', '') in al][:3]
-
-    # for the DQ and DP cases, alleles are matched (e.g. 'HLA-DQA10102-DQB10602')
-    # If two alleles are present is such a combi case, select that combi case as the target allele
-    target_alleles_matched = []
-    for al in target_alleles:
-        hits = 0
-        for part in [i.split('*')[-1] for i in allele_type]:
-            if part in al:
-                hits +=1
-        if hits == 2:
-            target_alleles_matched.append(al)
-    if len(target_alleles_matched) > 0:
-        target_alleles = target_alleles_matched
-
-    target_alleles = [i for i in target_alleles if i in all_netMHCpan_alleles]
-    # If there are no target alleles that occur in netMHCIIpan, but there is a mouse allele, use all mouse alleles
-    # that are supported by netMHCIIpan
-    if target_alleles == [] and any(al.startswith('H2') for al in allele_type):
-        target_alleles = [i for i in all_netMHCpan_alleles if i.startswith('H-')]
-
-    # If there is no target allele that occurs in netMHCIIpan, just use the standard DRB1_0101
-    if target_alleles == []:
-        target_alleles = ['DRB1_0101']
-
-    target_alleles_str = ','.join(target_alleles)
-
-    # Setup files
-    netmhciipan = PANDORA.PANDORA_path + '/../netMHCIIpan-4.0/netMHCIIpan'
-    infile = PANDORA.PANDORA_path + '/../netMHCIIpan-4.0/tmp/%s_%s_%s.txt' %(
-        peptide, target_alleles[0], datetime.today().strftime('%Y%m%d_%H%M%S'))
-    outfile = PANDORA.PANDORA_path + '/../netMHCIIpan-4.0/tmp/%s_%s_%s_prediction.txt' %(
-        peptide, target_alleles[0], datetime.today().strftime('%Y%m%d_%H%M%S'))
-
-    # Write peptide sequence to input file for netMHCIIpan
-    with open(infile, 'w') as f:
-        f.write(peptide)
-
-    try:
-        # run netMHCIIpan
-        os.system('%s -f %s -inptype 1 -a %s > %s' % (netmhciipan, infile, target_alleles_str, outfile))
+    
+    # Retrieves the enviroment of the netMHCIIpan script and the path of the directory where this script is in
+    if os.getenv('netMHCpan', default=None):
+        netmhcpan_file_path = os.getenv('netMHCIIpan', default=None)
+        netmhcpan_path = os.path.dirname(netmhcpan_file_path) 
 
-        # Get the output from the netMHCIIpan prediction
-        # {allele: (offset, core, core_reliability, score_EL, %rank_EL)}
-        pred = {}
-        with open(outfile) as f:
+        all_netMHCpan_alleles = []
+        with open(netmhcpan_path + '/data/allele.list') as f:
```

Comment:
Please use os.path.join() to define any path in these functions. Also paths that you have not changed yourself, it will improve the code interoperability. Have a look at the whole function and see where you can fix it.

Reply:
I did this in commit [2c170ad](https://github.com/X-lab-3D/PANDORA/pull/196/commits/2c170ad49032e2083b5f232303f79e3afeb1ac5b)